spec/datadog/tracing/integration_spec.rb

```diff
@@ -312,12 +338,7 @@ def agent_receives_span_step3(previous_success)
         c.tracing.priority_sampling = priority_sampling if priority_sampling
       end
 
-      # Capture trace segments as they are about to be serialized
-      allow_any_instance_of(Datadog::Transport::Traces::Transport)
-        .to receive(:send_traces).and_wrap_original do |function, traces|
-          trace_segments.concat(traces)
-          function.call(traces)
-        end
+      WebMock.enable!
```

Comment:
These `WebMock` changes are due to a flaky test, `sets the Datadog-Client-Computed-Top-Level header to a non-empty value`, that can fail if `WebMock` was disabled by a previous test case.

This is unrelated to this PR, but became very prominent when I added more WebMock related tests to this file.

The fix is in this PR because CI broke in this very branch due to this flakiness.

Reply:
May I suggest moving the `enable!`/`disable!` to the `context` that actually uses it, rather than enabling it for the whole file?